src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -558,31 +558,34 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
             var cookie2 = new KeyValuePair<string, string>(".AspNetCore.Session", "RAExEmXpoCbueP_QYM");
             var cookie3 = new KeyValuePair<string, string>("name", "value");
 
-            string url = string.Format(
-                "http://httpbin.org/cookies/set?{0}={1}&{2}={3}&{4}={5}",
-                cookie1.Key,
-                cookie1.Value,
-                cookie2.Key,
-                cookie2.Value,
-                cookie3.Key,
-                cookie3.Value);
-
-            var handler = new HttpClientHandler()
+            await CreateServerAsync(async (handler, client, server, url) =>
             {
-                AllowAutoRedirect = false
-            };
+                Task<HttpResponseMessage> getResponse = client.GetAsync(url);
 
-            using (var client = new HttpClient(handler))
-            using (HttpResponseMessage response = await client.GetAsync(url))
-            {
-                Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
-                CookieCollection cookies = handler.CookieContainer.GetCookies(new Uri(url));
+                await AcceptSocketAsync(server, async (s, stream, reader, writer) =>
+                {
+                    while (!string.IsNullOrEmpty(reader.ReadLine())) ;
+                    await writer.WriteAsync(
+                        $"HTTP/1.1 200 OK\r\n" +
```

Comment:
RFC 723x are considered the later versions since 2616 is deprecated.
https://tools.ietf.org/html/rfc7231

Section 7.1.1.2:
>   An origin server MUST NOT send a Date header field if it does not
   have a clock capable of providing a reasonable approximation of the
   current instance in Coordinated Universal Time.  An origin server MAY
   send a Date header field if the response is in the 1xx
   (Informational) or 5xx (Server Error) class of status codes.  An
   origin server MUST send a Date header field in all other cases.

So, in generally, it's always a good idea to have it.